### PR TITLE
Resolve #112: Add support for http HEAD method to RestClient

### LINKED ISSUE #112: Add support for http HEAD method to RestClient

### PR DESCRIPTION
**Pull request by saalistaja**
All current GET endpoints on the test controller support also HEAD requests.

### CODE REVIEW

**Comment by saalistaja** [comment]
I was considering removing the response body type from the methods (as HEAD request has no response body), but it stayed there to keep same patterns for all HTTP methods.

File: README.md [hunk 1]
```diff
@@ -162,6 +162,10 @@ Once the rest client is initialized, you can use the following methods. Each met
 - `patch` - a blocking PATCH call with a generic request / response
 - `patchNonBlocking` - a non-blocking PATCH call with a generic request / response with `onSuccess` and `onError` consumers
 - `patchObject` - a blocking PATCH call with `ObjectRequest` / `ObjectResponse`
+- `head` - a blocking PATCH call with a generic request
```
Comment (banterCZ) [inline README.md:165]:
```suggestion
- `head` - a blocking HEAD call with a generic request
```

File: README.md [hunk 2]
```diff
@@ -162,6 +162,10 @@ Once the rest client is initialized, you can use the following methods. Each met
...
+- `headNonBlocking` - a non-blocking PATCH call with a generic request with `onSuccess` and `onError` consumers
```
Comment (banterCZ) [inline README.md:166]:
```suggestion
- `headNonBlocking` - a non-blocking HEAD call with a generic request with `onSuccess` and `onError` consumers
```

File: README.md [hunk 3]
```diff
@@ -162,6 +162,10 @@ Once the rest client is initialized, you can use the following methods. Each met
...
+- `headObject` - a blocking PATCH call with `ObjectRequest`
```
Comment (banterCZ) [inline README.md:167]:
```suggestion
- `headObject` - a blocking HEAD call with `ObjectRequest`
```

**Review by banterCZ — CHANGES_REQUESTED** [review]
Looks good, just typos in the documentation.

File: README.md (hunk 1 above)
Comment (saalistaja) [reply README.md]: ouch, thanks

**Review by banterCZ — APPROVED** [review]

**Review by romanstrobl — APPROVED** [review]
Useful addition. We can keep the empty response classes, this is fine.

**Pull request merged**